### PR TITLE
common: configure and fix kaustinen4 verkle testnet sync

### PR DESCRIPTION
**Pull request by g11tech**
1. run ethereumjs: `DEBUG=ethjs,vm:*,evm:*,statemanager:verkle* npm run client:start:ts -- --dataDir /data/teme --network kaustinen4 --rpcEngine --rpcEngineAuth false`
2. run lodestar-quickstart (branch from this PR unless merged: https://github.com/ChainSafe/lodestar-quickstart/pull/39 ) : `./setup.sh --dataDir k4data --network kaustinen4 --justCL`
  Comment Line 26 and uncomment Line 25 to prevent lodestar from checkpoint syncing 

UPDATE:
Kautinen4 sync works upto block 373, at block 374 an execution witness seems to be missing for extcodesize opcode evaluation on transaction 3, escalated the issue:
```
Error: Missing execution witness for address=0x00000000000055600060ce55b5600060b955367f versionKey=0x48ac8240e218444ddffaabcc3270229064c0e918ee3578155aca0636e822db00
```

fixes:
 - [x] fix the vmexecution  for verkle stateless init
 - [x] track and ignore the BLOCKHASH witness to avoid k3 blockhash issues
 - [x] don't deduct the to and from access fees (block11)
 - [x] kaustinen3 block 15 : 
   It seems that a contract which was created, its code read accesses were charged which it shouldn't have been charged as the code was ready from the data, escalled to Guilluame for clarification, fixed in kaustinen4 
- [x] debug and fix the code chunking boundary conditions
- [x] debug and fix the kaustinen4 failing block 368
- [x] debug the failing block374 and escallate the issue to Guilluame

### CODE REVIEW

File: packages/client/test/rpc/engine/kaustinen4.spec.ts [hunk 1]
```diff
@@ -7,13 +7,30 @@ import genesisJSON from '../../testdata/geth-genesis/kaustinen4.json'
...
 import type { HttpClient } from 'jayson/promise'
 const genesisVerkleStateRoot = '0x382960711d9ccf58b9db20122e2253eb9bfa99d513f8c9d4e85b55971721f4de'
 const genesisVerkleBlockHash = '0x8493ed97fd4314acb6ed519867b086dc698e25df37ebe8f2bc77313537710744'
 
+/**
+ * One can run this test in two formats:
+ *   1. On the saved blocks, comma separated which are limited (345,375,368,467)
+ *      `TEST_SAVED_NUMBERS=345,375,368,467 npx vitest run test/rpc/engine/kaustinen4.spec.ts`
+ *   2. Directly pull slots from a kaustinen beacon url
+ *     `TEST_ONLINE_SLOTS=345,353..360 PEER_BEACON_URL=https://beacon.verkle-gen-devnet-4.ethpandaops.io npx vitest run test/rpc/engine/kaustinen4.spec.ts`
```
Comment (holgerd77) [inline packages/client/test/rpc/engine/kaustinen4.spec.ts:21]: This is so cool, just tested this, works like a charm!!! 🔥 👍 🙂

File: packages/common/src/interfaces.ts [hunk 2]
```diff
@@ -72,6 +72,7 @@ export interface StateManagerInterface {
   modifyAccountFields(address: Address, accountFields: AccountFields): Promise<void>
   putContractCode(address: Address, value: Uint8Array): Promise<void>
   getContractCode(address: Address): Promise<Uint8Array>
+  getContractCodeSize(address: Address): Promise<number>
```
Comment (holgerd77) [inline packages/common/src/interfaces.ts:75]: This is a breaking change, we need to make this optional.

**Review by holgerd77 — COMMENTED** [review]
Ok, I am through, one small thing, otherwise looks good!

(will fully approve once tests pass)

File: packages/common/src/interfaces.ts (hunk 2 above)
Comment (g11tech) [reply packages/common/src/interfaces.ts]: done :pray:

**Review by holgerd77 — APPROVED** [review]
Ok, have deactivated the devp2p tests until we have these reliable again.

Guess we can take this in, right? Will approve and merge. 🙂
